### PR TITLE
feat: restore PO-Revision-Date but with fixed value

### PR DESCRIPTION
**Pull request by shadinaif**
External packages are forcing validation on `PO-Revision-Date`. Therefore, we are restoring both `POT-Creation-Date` and `PO-Revision-Date` but setting a fixed value for them to avoid having a lot of false (git diff) lines. Note: (2023-06-13 is Palm release date)

This is a fix for a defect reported in `credentials` repository: https://github.com/openedx/i18n-tools/issues/131

Reverts part of the work done here https://github.com/openedx/i18n-tools/pull/129

This contribution is part of the [FC-0012 project](https://openedx.atlassian.net/l/cp/XGS0iCcQ) which is sparked by the [Translation Infrastructure update OEP-58](https://open-edx-proposals.readthedocs.io/en/latest/architectural-decisions/oep-0058-arch-translations-management.html#specification).

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @shadinaif!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

**Comment by shadinaif** [comment]
Ready for review @OmarIthawi @brian-smith-tcril

File: i18n/extract.py
Comment (OmarIthawi) [inline i18n/extract.py:319]: Please add `POT-Creation-Date` and put it on the top -- if the order matters.

Also, some tools may check for [timestamp epoch 0](https://www.unixtimestamp.com/).

Use any date you like that's relatively recent.

Comment (shadinaif) [reply i18n/extract.py]: Done

**Comment by OmarIthawi** [comment]
Thanks @shadinaif!! Some tests are failing. Would you mind taking a look?

Also I think it makes sense to have an integration test case which runs the following two commands on a sample repo test folder:

```
cd tests/data/mock-django-app
# Needs some python files to generate languages for
i18n_tool extract
i18n_tool validate
```

I think this will ensure that this error will not happen again, what do you think?

File: i18n/extract.py
Comment (OmarIthawi) [inline i18n/extract.py:320]: One of them has an incorrect timezone. Please make the consistent.

```suggestion
        'POT-Creation-Date': '2023-06-13 08:00+0000',
        'PO-Revision-Date': '2023-06-13 09:00:00.000000',
```

File: tests/test_integration.py
Comment (OmarIthawi) [inline tests/test_integration.py:25]:
```suggestion
    def verify_via_validate_command(self, check_all):
```

File: i18n/extract.py
Comment (shadinaif) [inline i18n/extract.py:321]: if you restore this line @OmarIthawi , the integration test will fail. `pofile.metadata.pop('PO-Revision-Date', None)`

Restoring the creation date will not fail the test because it's not being checked by `validate` anyway

File: tests/test_integration.py
Comment (OmarIthawi) [inline tests/test_integration.py:42]:
```suggestion
            # Extract is called by the `@test_extract.perform_extract_with_options()` helper.
            result = self.verify_via_validate_command(check_all)
```

**Review by OmarIthawi — COMMENTED** [review]
Thanks @shadinaif. Few more minor notes and it should be ready.

File: i18n/extract.py
Comment (shadinaif) [reply i18n/extract.py:320]: This is the original format by `django-admin makemessages` I promise 🥲

File: i18n/extract.py
Comment (OmarIthawi) [reply i18n/extract.py:321]: > Restoring the creation date will not fail the test because it's not being checked by validate anyway

Please feel free to restore only what's required by `validate`. I wouldn't rely on my intuition for this obscure validation part.

Comment (shadinaif) [reply i18n/extract.py:321]: I think we're good, let's keep them both

File: tests/test_integration.py
Comment (shadinaif) [reply tests/test_integration.py]: done

File: tests/test_integration.py
Comment (shadinaif) [reply tests/test_integration.py]: done

File: i18n/extract.py
Comment (OmarIthawi) [reply i18n/extract.py:320]: Add a comment please inline:

```suggestion
        # The dates has been copied as-is from `django-admin makemessages` results.
        # The timezone format differs, and that's what `makemessages` outputs.
        'POT-Creation-Date': '2023-06-13 08:00+0000',
        'PO-Revision-Date': '2023-06-13 09:00:00.000000',
```

**Review by OmarIthawi — APPROVED** [review]
Thanks a lot @shadinaif! I've tested it with the following:

```
$ git clone git@github.com:Zeit-Labs/i18n-tools.git --branch=shadinaif/restore-revision-date
$ cd credentials
$ make requirements  # credentials requirements
$ pip install -e ../i18n-tools
$ make extract_translations
```

Output seems to be working okay!
```
$ make validate_translations
.....
INFO:i18n.validate:No problems found in /home/omar/work/openedx/credentials/credentials/conf/locale/es_AR/LC_MESSAGES/django.po
INFO:i18n.execute:msgfmt -c -o /dev/null es_AR/LC_MESSAGES/djangojs.po
INFO:i18n.validate:No problems found in /home/omar/work/openedx/credentials/credentials/conf/locale/es_AR/LC_MESSAGES/djangojs.po
omar@antilop:credentials $ 
...
```

File: i18n/extract.py
Comment (shadinaif) [reply i18n/extract.py]: I made them consistent already, thank you for your prompt reviews @OmarIthawi !

**Comment by awais786** [comment]
@OmarIthawi any thing left in this PR ? waiting for new release to unblock credentials.

**Comment by OmarIthawi** [comment]
Thanks for checking @awais786. We need someone to test it on the credentials repo, I couldn't get to this in a timely manner.

Justin [was going to test it](https://github.com/openedx/i18n-tools/issues/131#issuecomment-1746768768), but please feel free to do so.

If it's good, I can merge and cut a new release.

**Comment by awais786** [comment]
@OmarIthawi https://github.com/openedx/credentials/pull/2221 Please check this branch.

**Comment by awais786** [comment]
@OmarIthawi https://github.com/openedx/credentials/pull/2221 plz check this. Seems trans are working,

**Review by awais786 — APPROVED** [review]

**Comment by awais786** [comment]
now credentials failing with black issue. 

black --check .
```
would reformat /home/runner/work/credentials/credentials/src/edx-i18n-tools/i18n/branch_cleanup.py
would reformat /home/runner/work/credentials/credentials/src/edx-i18n-tools/i18n/__init__.py
would reformat /home/runner/work/credentials/credentials/src/edx-i18n-tools/i18n/changed.py
would reformat /home/runner/work/credentials/credentials/src/edx-i18n-tools/i18n/converter.py
would reformat /home/runner/work/credentials/credentials/src/edx-i18n-tools/i18n/config.py
would reformat /home/runner/work/credentials/credentials/src/edx-i18n-tools/i18n/execute.py
would reformat /home/runner/work/credentials/credentials/src/edx-i18n-tools/i18n/dummy.py
would reformat /home/runner/work/credentials/credentials/src/edx-i18n-tools/i18n/extract.py
would reformat /home/runner/work/credentials/credentials/src/edx-i18n-tools/i18n/generate.py
would reformat /home/runner/work/credentials/credentials/src/edx-i18n-tools/i18n/main.py
would reformat /home/runner/work/credentials/credentials/src/edx-i18n-tools/i18n/segment.py
would reformat /home/runner/work/credentials/credentials/src/edx-i18n-tools/i18n/transifex.py
would reformat /home/runner/work/credentials/credentials/src/edx-i18n-tools/scripts/podup.py
would reformat /home/runner/work/credentials/credentials/src/edx-i18n-tools/setup.py
would reformat /home/runner/work/credentials/credentials/src/edx-i18n-tools/i18n/validate.py
would reformat /home/runner/work/credentials/credentials/src/edx-i18n-tools/tests/__init__.py
would reformat /home/runner/work/credentials/credentials/src/edx-i18n-tools/tests/test_config.py
would reformat /home/runner/work/credentials/credentials/src/edx-i18n-tools/tests/test_changed.py
would reformat /home/runner/work/credentials/credentials/src/edx-i18n-tools/tests/test_converter.py
would reformat /home/runner/work/credentials/credentials/src/edx-i18n-tools/tests/test_dummy.py
would reformat /home/runner/work/credentials/credentials/src/edx-i18n-tools/tests/test_extract.py
would reformat /home/runner/work/credentials/credentials/src/edx-i18n-tools/tests/test_generate.py
would reformat /home/runner/work/credentials/credentials/src/edx-i18n-tools/tests/test_integration.py
would reformat /home/runner/work/credentials/credentials/src/edx-i18n-tools/tests/test_transifex.py
would reformat /home/runner/work/credentials/credentials/src/edx-i18n-tools/tests/test_segment.py
would reformat /home/runner/work/credentials/credentials/src/edx-i18n-tools/tests/test_validate.py

```

**Review by awais786 — APPROVED** [review]

**Comment by justinhynes** [comment]
I wasn't able to get to this yesterday. I tried it out this morning and I was able to validate translations locally without any further issues. Thanks for the work on this!

**Review by justinhynes — APPROVED** [review]

**Comment by awais786** [comment]
@justinhynes it fails on ci with black issue. You can check my PR. https://github.com/openedx/credentials/pull/2221

**Comment by awais786** [comment]
@OmarIthawi plz release new version.

**Comment by justinhynes** [comment]
@awais786 That's interesting... Black shouldn't be trying to reformat the files of an external library. 🤔 Worst case maybe we just need to update `pyproject.toml` to ensure it's properly excluding directories? Or maybe it's a symptom of how the library was installed? 

Either way, that problem is definitely on the Credentials side and I'm sure we can figure out a way to work around it.

**Comment by awais786** [comment]
> @awais786 That's interesting... Black shouldn't be trying to reformat the files of an external library. 🤔 Worst case maybe we just need to update `pyproject.toml` to ensure it's properly excluding directories? Or maybe it's a symptom of how the library was installed?
> 
> Either way, that problem is definitely on the Credentials side and I'm sure we can figure out a way to work around it.

I have already fixed https://github.com/openedx/credentials/pull/2221

**Comment by openedx-webhooks** [comment]
@shadinaif 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/i18n-tools&entry.867055334=https://github.com/openedx/i18n-tools/pull/140&entry.1484655318=https://github.com/shadinaif&entry.752974735=2023-09-27+12:56&entry.1917517419=2023-10-06+06:09&entry.2133058324=Yes) so we can improve your experience in the future.

**Comment by omar-nelc** [comment]
@awais786 thanks for the review and the release cut. I was about to merge and cut the release, but you've beat me to it.

Thanks again for the thorough report! It's making this tool much better :)

**Comment by awais786** [comment]
@OmarIthawi Thanks for the quick fix.

**Pull request merged**
